codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/auth/AuthSchemeParametersGenerator.kt

```diff
@@ -103,4 +70,62 @@ class AuthSchemeParametersGenerator : AbstractConfigGenerator() {
             render(codegenCtx, props, writer)
         }
     }
+
+    override fun renderAdditionalMethods(ctx: CodegenContext, props: List<ConfigProperty>, writer: KotlinWriter) {
+        writer.write("")
+        renderToString(ctx, props, writer)
+        writer.write("")
+        renderEquals(ctx, props, writer)
+        writer.write("")
+        renderHashCode(ctx, props, writer)
+        writer.write("")
+        renderCopy(ctx, props, writer)
+    }
+
+    private fun renderEquals(ctx: CodegenContext, props: List<ConfigProperty>, writer: KotlinWriter) {
+        writer.withBlock("$visibility override fun equals(other: Any?): Boolean {", "}") {
+            write("if (this === other) return true")
+            write("if (other !is #T) return false", getSymbol(ctx.settings))
+            props.forEach { prop ->
+                write("if (this.#1L != other.#1L) return false", prop.propertyName)
+            }
+            write("return true")
+        }
+    }
+    private fun renderToString(ctx: CodegenContext, props: List<ConfigProperty>, writer: KotlinWriter) {
+        writer.withBlock("$visibility override fun toString(): String = buildString {", "}") {
+            write("append(\"#L(\")", getSymbol(ctx.settings).name)
+            props.forEachIndexed { idx, prop ->
+                write("""append("#1L=$#1L#2L")""", prop.propertyName, if (idx < props.size - 1) "," else ")")
+            }
+        }
+    }
+
+    private fun renderHashCode(ctx: CodegenContext, props: List<ConfigProperty>, writer: KotlinWriter) {
+        writer.withBlock("public override fun hashCode(): Int {", "}") {
```

Comment:
Should this always be public or also controlled by `visibility`? Same for copy function